### PR TITLE
Create ARO cluster with pre-existing network security group. 

### PR DESCRIPTION
**Pull request by mgahagan73**
Create ARO cluster with pre-existing network security group.

### CODE REVIEW

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-aro

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure-aro

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure-aro

**Comment by MayXuQQ** [comment]
@jinyunma could you help review the pr? thanks

File: ci-operator/step-registry/aro/provision/nsg/aro-provision-nsg-commands.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+CLUSTER=${CLUSTER:="${NAMESPACE}-${UNIQUE_HASH}"}
+RESOURCEGROUP=${RESOURCEGROUP:=$(cat "${SHARED_DIR}/resourcegroup")}
+VNET=${VNET:=$(cat "${SHARED_DIR}"/vnet)}
+LOCATION=${LOCATION:=${LEASED_RESOURCE}}
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+MASTER_SUBNET_NAME=${MASTER_SUBNET_NAME:=$(grep controlPlaneSubnet ${SHARED_DIR}/customer_vnet_subnets.yaml | cut -d ":" -f 2 | tr -d "[:blank:]")}
+WORKER_SUBNET_NAME=${WORKER_SUBNET_NAME:=$(grep computeSubnet ${SHARED_DIR}/customer_vnet_subnets.yaml | cut -d ":" -f 2 | tr -d "[:blank:]")}
```
Comment (jianlinliu) [inline ci-operator/step-registry/aro/provision/nsg/aro-provision-nsg-commands.sh:19]:
```suggestion
MASTER_SUBNET_NAME=$(yq-go r ${SHARED_DIR}/customer_vnet_subnets.yaml 'platform.azure.controlPlaneSubnet')
WORKER_SUBNET_NAME=$(yq-go r ${SHARED_DIR}/customer_vnet_subnets.yaml 'platform.azure.computeSubnet')
```

File: ci-operator/step-registry/aro/provision/nsg/aro-provision-nsg-commands.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#echo "Installing oc binary"
+#curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/openshift-client-linux.tar.gz | tar zxvf - oc
+#chmod +x oc
+CLUSTER=${CLUSTER:="${NAMESPACE}-${UNIQUE_HASH}"}
```
Comment (jianlinliu) [inline ci-operator/step-registry/aro/provision/nsg/aro-provision-nsg-commands.sh:10]: The step does not declare `CLUSTER` env, so I think it should be:
```suggestion
CLUSTER="${NAMESPACE}-${UNIQUE_HASH}"
```

File: ci-operator/step-registry/aro/provision/nsg/aro-provision-nsg-commands.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+RESOURCEGROUP=${RESOURCEGROUP:=$(cat "${SHARED_DIR}/resourcegroup")}
```
Comment (jianlinliu) [inline ci-operator/step-registry/aro/provision/nsg/aro-provision-nsg-commands.sh:11]: The step does not declare `RESOURCEGROUP` env, so I think it should be:
```suggestion
RESOURCEGROUP=$(cat "${SHARED_DIR}/resourcegroup")
```

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure-aro

**Pull request merged**